cmd/deploy/validate_test.go

```diff
@@ -14,37 +14,80 @@
 package deploy
 
 import (
+	"os"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 )
 
-func Test_ValidateVars(t *testing.T) {
-
+func Test_setEnvVars(t *testing.T) {
 	var tests = []struct {
-		name        string
-		variables   []string
-		expectedErr bool
+		name            string
+		variables       []string
+		expectedErr     bool
+		expectedEnvVars []envKeyValue
 	}{
 		{
-			name:        "correct assingnament",
+			name:        "correct separator",
 			variables:   []string{"NAME=test"},
 			expectedErr: false,
+			expectedEnvVars: []envKeyValue{
+				{
+					key:   "Name",
+					value: "test",
+				},
+			},
+		},
+		{
+			name:            "bad separator",
+			variables:       []string{"NAME:test"},
+			expectedErr:     true,
+			expectedEnvVars: []envKeyValue{},
+		},
+		{
+			name:            "bad count",
+			variables:       []string{"=foo"},
+			expectedErr:     true,
+			expectedEnvVars: []envKeyValue{},
+		},
+		{
+			name:            "one bad",
+			variables:       []string{"first=one", "second=two", "third:three"},
+			expectedErr:     true,
+			expectedEnvVars: []envKeyValue{},
 		},
 		{
-			name:        "bas assingnament",
-			variables:   []string{"NAME:test"},
-			expectedErr: true,
+			name:        "multiples",
+			variables:   []string{"first=one", "second=two", "third=three"},
+			expectedErr: false,
+			expectedEnvVars: []envKeyValue{
+				{
+					key:   "first",
+					value: "one",
+				},
+				{
+					key:   "second",
+					value: "two",
+				},
+				{
+					key:   "third",
+					value: "three",
+				},
+			},
 		},
 	}
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			err := validateOptionVars(tt.variables)
+			t.Parallel()
```

Comment:
How does parallel testing and the os package behave together? Could it lead to problems? i.e. changing envvars for another test

Reply:
It can lead to problems and not just due to the parallelism. I think I'll need to use some of the newer env var testing facilities in go and possibly refactor a bit. I'm going to try it out.